test.py

```diff
@@ -27,6 +27,8 @@
     parser.add_argument('--fast',  action="store_true", help="Run only fast tests")
     parser.add_argument('--name',  action="store", help="Run only test whose name contains given string")
     parser.add_argument('--mode', choices=seastar_cmake.SUPPORTED_MODES, help="Run only tests for given build mode")
+    parser.add_argument('--build-root', action='store', default='build', type=str, help="The name of the build root build directoy: "
```

Comment:
allowing user to specify the build directory actually  ~~opens a Pandora's box~~ enables them to use "foo/bar" here. but we have an assumption hiding at https://github.com/scylladb/seastar/blob/eee093aeb915ecd8c3c4b60564b49e01e81e9b20/configure.py#L242 .

so, if we want to support the use case, probably want to use something like

```py
parser.add_argument('--build-root', action='store', default='build', type=pathlib.Path, help="...")
```

to ease the manipulating of the path, as `pathlib.Path` offers methods like `relative_to()`, which could be handy. because we would need to specify the source directory after `chdir` into the specified build directory, which could be "my-fix/clang-15/debug", and `ARGS` should be `['cmake', '-G', 'Ninja', '../../..']`, if you prefer keeping using relative path, which i recommend as well.

Reply:
Good catch.

I've used `os.path.relpath()` to do the generation of the relative path to "back out" from the build dir to the root repo dir. My understanding is that pathlib doesn't support that currently (generating `..` in paths).